### PR TITLE
[Doc] Move guide for multimodal model and other improvements

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR primarily moves the multimodal model guide under "Models" section for more visibility. Some other changes:

- Clean up the input processing and multimodal docs.
- Reduce nesting of "Developer Documentation" in the main page since it's a bit spammy.

### CODE REVIEW

File: docs/source/models/adding_model.rst [hunk 1]
```diff
@@ -44,23 +44,23 @@ Next, you need to rewrite the :meth:`~torch.nn.Module.forward` method of your mo
...
+    -     labels: Optional[torch.LongTensor] = None,
+    -     use_cache: Optional[bool] = None,
+    -     output_attentions: Optional[bool] = None,
+    -     output_hidden_states: Optional[bool] = None,
+    -     return_dict: Optional[bool] = None,
+    - ) -> Union[Tuple, CausalLMOutputWithPast]:
+    +     positions: torch.Tensor,
+    +     kv_caches: List[torch.Tensor],
+    +     attn_metadata: AttentionMetadata,
+    + ) -> Optional[SamplerOutput]:
```
Comment (ywang96) [inline docs/source/models/adding_model.rst:63]: Was this necessary?

File: docs/source/models/adding_multimodal_model.rst [hunk 2]
```diff
@@ -38,14 +38,14 @@ As usual, follow :ref:`these steps <adding_a_new_model>` to implement the model
...
-      +   pixel_values: torch.Tensor,
-      ) -> SamplerOutput:
+        def forward(
+            self,
+            input_ids: torch.Tensor,
+            positions: torch.Tensor,
+            kv_caches: List[torch.Tensor],
+            attn_metadata: AttentionMetadata,
+      +     pixel_values: torch.Tensor,
+        ) -> SamplerOutput:
```
Comment (ywang96) [inline docs/source/models/adding_multimodal_model.rst:48]: Ditto

**Review by ywang96 — APPROVED** [review]

File: docs/source/models/adding_model.rst (hunk 1 above)
Comment (DarkLight1337) [reply docs/source/models/adding_model.rst:63]: The previous indentation was a bit hard to read as the original and updated Python code were offset from each other.

**Comment by DarkLight1337** [comment]
I've renamed the guide to "Enabling Multimodal Inputs" to avoid overlap with the base "Adding a New Model". PTAL again.

File: docs/source/models/enabling_multimodal_inputs.rst [hunk 3]
```diff
@@ -1,26 +1,21 @@
...
+This document walks you through the steps to extend a vLLM model so that it accepts :ref:`multi-modal <multi_modality>` inputs.
 
-.. note::
-    The complexity of adding a new model depends heavily on the model's architecture.
-    The process is considerably straightforward if the model shares a similar architecture with an existing model in vLLM.
-    However, for models that include new operators (e.g., a new attention mechanism), the process can be a bit more complex.
-
-.. tip::
-    If you are encountering issues while integrating your model into vLLM, feel free to open an issue on our `GitHub <https://github.com/vllm-project/vllm/issues>`_ repository.
-    We will be happy to help you out!
```
Comment (ywang96) [inline docs/source/models/enabling_multimodal_inputs.rst:15]: Maybe keep some of this?

**Review by ywang96 — APPROVED** [review]

File: docs/source/models/enabling_multimodal_inputs.rst (hunk 3 above)
Comment (DarkLight1337) [reply docs/source/models/enabling_multimodal_inputs.rst:15]: I've already linked users to the base "Adding a New Model" which includes the same content.

**Pull request merged**